src/cmd/ksh93/tests/basic.sh

```diff
@@ -536,7 +538,9 @@ float sec=SECONDS
 (( (SECONDS-sec) < .7 ))  && err_exit '. script does not restore output redirection with eval'
 
 file=$tmp/foobar
-builtin cat
+echo "TODO: Skipping call to builtin cat"
```

Comment:
The existing Nmake based build/test enables the builtin `mktemp` command which isn't present in the Meson built `ksh`. That leads to most of the tests failing with this error because they don't pass an explicit template and the BSD version of the command doesn't provide a default template:

```
mktemp: option requires an argument -- t
```

We could change all the test scripts by replacing `mktemp -dt` with `mktemp -dt tmp.XXXXXXXXXX` but it's probably better to figure out why we're not getting the builtin versions of these external commands aren't available.

Reply:
The builtins are present under ```src/lib/libcmd``` and I have not started compiling them yet. We should investigate which one of the builtins we should keep supporting.

Reply:
FYI, I edited all of the test scripts and made the aforementioned change to the `mktemp` invocations. That allows all of the tests to run. On second thought I think we should do that. We can always add explicit tests of the builtin `mktemp` if they don't already exist. I don't see any reason for the test harness to depend on its behavior since simply providing an explicit template works on BSD and Linux.

Reply:
> We should investigate which one of the builtins we should keep supporting.

There are people watching this project which feel quite strongly that all existing behavior _must be retained_. See, for example, the discussion about the obscure math functions that I will bet $100 have never been used in a ksh program. They will no doubt argue that all of the existing builtin versions of commands (that are externals in every other shell) must remain available as builtins. I'm ambivalent about it.

Consider the `realpath` command. The BSD and GNU versions are radically different except for their default behavior (i.e., when invoked with no flags). And some platforms supported by fish-shell don't have that command. So what the fish project decided to do was detect at run time whether a `grealpath` (in case the GNU version was installed under that name) exists and alias it to `realpath` if it does, if not does `realpath` exist, and if not then fallback to a fish builtin that takes no flags and implements only the default behavior of the external command (had it been found).

In general I think it is really dangerous to automatically expose a builtin with the same name as a widely available  external command such as `cat` or `mktemp`. But this is a discussion we should have in an issue rather than a PR.

Reply:
Please read the `builtin` documentation - at least what `builtin --man` shows. So binding a builtin command to SH_CMDLIB_DIR does not cause any collisions or unexpected behavior. If the user has $SH_CMDLIB_DIR (which is per default `/opt/ast/bin`) at the beginning of its PATH, he is asking exactly for this builtin, and exactly this enhanced/platform compatible/... behavior is expected. If not, it gets not used at all. So there is no need to kill builtins like cat or realpath.